docs/UserGuide/3-Server/6-MQTT Protocol.md

```diff
@@ -0,0 +1,82 @@
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+-->
+# MQTT Protocol
+
+[MQTT](http://mqtt.org/) is a machine-to-machine (M2M)/"Internet of Things" connectivity protocol.
+It was designed as an extremely lightweight publish/subscribe messaging transport.
+It is useful for connections with remote locations where a small code footprint is required and/or network bandwidth is at a premium.
+
+IoTDB supports the MQTT v3.1(an OASIS Standard) protocol.
+IoTDB server includes a built-in MQTT service that allows remote devices send messages into IoTDB server directly.
+
+<img style="width:100%; max-width:800px; max-height:600px; margin-left:auto; margin-right:auto; display:block;" src="https://user-images.githubusercontent.com/6711230/78357432-0c71cf80-75e4-11ea-98aa-c43a54d469ce.png">
+
+
+## Built-in MQTT Service
+The Built-in MQTT Service provide the ability of direct connection to IoTDB through MQTT. It listen the publish messages from MQTT clients
+ and then write the data into storage immediately. 
+The messages payload can be format to events by `PayloadFormatter` which loaded by java SPI, and the default implementation is `JSONPayloadFormatter`.
+
+<img style="width:100%; max-width:800px; max-height:600px; margin-left:auto; margin-right:auto; display:block;" src="https://user-images.githubusercontent.com/6711230/78357469-1bf11880-75e4-11ea-978f-a53996667a0d.png">
+
+## MQTT Configurations
+The IoTDB MQTT service load configurations from `${IOTDB_HOME}/${IOTDB_CONF}/iotdb-engine.properties` by default.
+
+Configurations are as following:
+
+| NAME        | DESCRIPTION           | DEFAULT  |
+| ------------- |:-------------:|:------:|
+| enable_mqtt_service      | whether to enable the mqtt service | true |
+| mqtt_host      | the mqtt service binding host | 0.0.0.0 |
+| mqtt_port      | the mqtt service binding port    |   1883 |
+| mqtt_handler_pool_size | the handler pool size for handing the mqtt messages      |    1 |
+| mqtt_payload_formatter | the mqtt message payload formatter     |    json |
+
+
+## Examples
+The following is an example which a mqtt client send messages to IoTDB server.
+
+ ```java
+        MQTT mqtt = new MQTT();
+        mqtt.setHost("127.0.0.1", 1883);
+        mqtt.setUserName("root");
+        mqtt.setPassword("root");
+
+        BlockingConnection connection = mqtt.blockingConnection();
+        connection.connect();
+
+        Random random = new Random();
+        for (int i = 0; i < 10; i++) {
+            Map<String,Object> tuple = new HashMap();
+            tuple.put("device", "root.sg.d1");
+            tuple.put("timestamp", System.currentTimeMillis());
+            tuple.put("measurements", "s1");
+            tuple.put("values", random.nextDouble());
+
+            String payload = JSON.toJSONString(tuple);
+            connection.publish("root.sg.d1.s1", payload.getBytes(), QoS.AT_LEAST_ONCE, false);
```

Comment:
Is the first parameter the clientID, right? If so, change the example as "clientID1" is better.

Reply:
The first parameter isn't clientID. it's the mqtt topic which is corresponding to the IoTDB timeseries.

Reply:
I added some documents to make it clear.